lib/bolt/inventory/group2.rb

```diff
@@ -190,25 +200,28 @@ def add_target(target)
 
         target['config'] = config_only_plugin(target['config'])
 
-        unless target.include?('alias')
-          return
-        end
-
-        aliases = target['alias']
-        aliases = [aliases] if aliases.is_a?(String)
-        unless aliases.is_a?(Array)
-          msg = "Alias entry on #{target['name']} must be a String or Array, not #{aliases.class}"
-          raise ValidationError.new(msg, @name)
-        end
+        if target.include?('alias')
+          aliases = target['alias']
+          aliases = [aliases] if aliases.is_a?(String)
+          unless aliases.is_a?(Array)
+            msg = "Alias entry on #{target['name']} must be a String or Array, not #{aliases.class}"
+            raise ValidationError.new(msg, @name)
+          end
 
-        aliases.each do |alia|
-          raise ValidationError.new("Invalid alias #{alia}", @name) unless alia =~ NAME_REGEX
+          aliases.each do |alia|
+            raise ValidationError.new("Invalid alias #{alia}", @name) unless alia =~ NAME_REGEX
 
-          if (found = @aliases[alia])
-            raise ValidationError.new(alias_conflict(alia, found, target['name']), @name)
+            if (found = @aliases[alia])
+              raise ValidationError.new(alias_conflict(alia, found, target['name']), @name)
+            end
+            @aliases[alia] = target['name']
           end
-          @aliases[alia] = target['name']
         end
+
+        raise ValidationError.new("Node #{target} does not have a name", @name) unless target['name']
+        t_name = target['name']
```

Comment:
I think this line can just be inline on 224:
```
@targets[target['name']] = target
```

Reply:
This might make more sense with the following two lines
```
        t_name = target['name']
        target.delete('name') if name_from_uri
        @targets[t_name] = target
```
if we set the name from the URI (in line 166) which will be the target name and is needed to keep track of targets in current group. When we later call `create_target` we need to set a safe_name. If we already have artificially assigned the name as the target URI, the algorithm in `creat_target` that decides when we need a safe name will be fooled. So the idea is to save the name, delete `target['name']` from the data (because we artificially populated it) and then assign to the group's `@targets` variable.